caps/types.go

```diff
@@ -65,3 +63,18 @@ func (t *Type) Validate(c *Capability) error {
 	}
 	return nil
 }
+
+// TypeLookupFn aids in looking up a Type by name
+type TypeLookupFn func(name string) *Type
+
+// MarshalJSON encodes a Type object as the name of the type.
+func (t *Type) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.Name)
```

Comment:
Not clear yet if that's what we really want here, or whether we want an actual objects with some fields. Seems fine for now.